### ISSUE TITLE
Html entities

### ISSUE THREAD

**Issue by nezih94** [noties/Markwon#127]
* **Markwon version**: _{3.0.1}_

1. Markwon unescapes html entities in a string, converting "\&gt;" to ">". But further processing doesn't happen and ">" character stays as is instead of creating a quote spannable out of it.
2. Try this string with Markwon : `&gt; This is a quote`

Another problem is markwon doesn't unescape an html entity at all if it's inside a code block / inline code.

**Comment by noties** [comment]
Hello @nezih94 !

**TL;DR**: it works absolutely as it should.

HTML escaped entities are here for exactly this reason: display special character without syntax-parsing. So, when you use `&gt;` you want to display actual `>` character without it being considered a part of HTML markup syntax.

Consider this input: `&gt; This is a quote`:

&gt; This is a quote

Or try to write some HTML markup: `&lt;span color="#F00"&gt;red&lt;/span&gt;`

&lt;span color="#F00"&gt;red&lt;/span&gt;

Code blocks on the other hand (HTML `<code>`) are special blocks that do not process the input and display it as it is (thus there is no need to escape symbols):

```
&lt;span color="#F00"&gt;red&lt;/span&gt;
```

```
<span color="#F00">red</span>
```

All of the above is not specific to `Markwon` library or even `commonmark` markdown specification (as you can see even Github markdown render follows it). So, there is nothing to fix.

By the way, you can use [this tool](https://spec.commonmark.org/dingus/?text=%0A%26gt%3B%20This%20is%20not%20a%20quote%0A%0A%3E%20This%20is%0A) to evaluate the markdown. This is not 100% compatible with `Markwon` (mainly due to HTML part), but it can give you a general picture of how things should be working.
